update README Carthage support to specify macOS issues (#160)
DevCycleHQ/ios-client-sdk: README.md

### Carthage

> **WARNING: Carthage Support is currently not working correctly for macOS, we are investigating soutions.**

Reviewer: ```suggestion
**WARNING: MacOS development with Carthage is currently not supported with DevCycle.**
```
